csrc/scheduler/matmul_heuristic.h

```diff
@@ -41,10 +41,18 @@ class MatmulParams : public HeuristicParams {
     // greater than one. Otherwise it is ignored.
     int smem_circular_buffer_stage = 2;
 
+    // The circular buffering prefetch distance will be set to
+    //   smem_circular_buffer_stage - smem_circular_buffer_prefetch_gap
+    // This value must be positive since the prefetch distance must be strictly
```

Comment:
Should we enforce this condition in `setUpCircularBuffering`?
e.g., 
```cpp
NVF_THROW(smem_circular_buffer_prefetch_gap >= 0 && smem_circular_buffer_prefetch_gap < smem_circular_buffer_stage, "smem_circular_buffer_prefetch_gap is ", smem_circular_buffer_prefetch_gap,
" but is expected to be positive and less than number of stages - ", smem_circular_buffer_stage);
```

Reply:
Checked that `prefix=0` does work. Added the assertion `0 < gap <= stages` in latest push.